src/robotide/parserlog/parserlog.py

```diff
@@ -97,29 +99,42 @@ def _log_message(self, log_event):
 
     def OnViewLog(self, event, show_tab=True):
         if not self._panel:
-            self._panel = _LogWindow(self.notebook, self._log)
+            self._panel = _LogWindow(self.notebook, self._log, self.title)
             self.notebook.SetPageTextColour(self.notebook.GetPageCount()-1, wx.Colour(255, 165, 0))
             self._panel.update_log()
-            self.register_shortcut('CtrlCmd-C', lambda e: self._panel.Copy())
-            self.register_shortcut(
-                 'CtrlCmd-A', lambda e: self._panel.SelectAll())
         if show_tab:
             self.notebook.show_tab(self._panel)
+        if self.is_focused():
+            self._register_shortcuts()
+
+    def OnTabChange(self, message):
+        if message.newtab == self.title:
+            self._register_shortcuts()
+        elif message.oldtab == self.title:
+            self.unregister_actions()
+
+    def is_focused(self):
+        return self.notebook.current_page_title == self.title
+
+    def _register_shortcuts(self):
+        self.register_shortcut('CtrlCmd-C', lambda e: self._panel.Copy())
+        self.register_shortcut('CtrlCmd-A', lambda e: self._panel.SelectAll())
 
```

Comment:
I managed to get Parser log receiving Ctrl-A even if tab was closed. It received Ctrl-A from kweditor and text editor (not working there). This is console outuput:
```
Traceback (most recent call last):
  File "C:\github\RIDE\src\robotide\ui\actiontriggers.py", line 338, in __call__
    action.act(event)
  File "C:\github\RIDE\src\robotide\action\action.py", line 89, in act
    self.action(event)
  File "C:\github\RIDE\src\robotide\parserlog\parserlog.py", line 121, in <lambda>
    self.register_shortcut('CtrlCmd-A', lambda e: self._panel.SelectAll())
  File "C:\github\RIDE\src\robotide\parserlog\parserlog.py", line 163, in SelectAll
    self._output.SetSelection(-1, -1)
RuntimeError: wrapped C/C++ object of type TextCtrl has been deleted
```
(Windows 10 with Python 2.7.16)

Reply:
Hm, good find. It seems that shortcuts are not unregistered when tab is closed. I've tried several workarounds, but they all fell short. I think `register_shortcut` is meant to be used only when there are menu entries involved. As such, let's use key events as much as possible from now on. These cannot create conflicts because they are bound to the object inside the tab.

Reply:
I've updated RIDE and Parser Log with the new method. Please check it out.
If everything is ok then should I proceed and update Text Editor using the same method?

Reply:
Parser log is a copy of RIDE Log. The Select All keyboard shortcut is working on both windows, but the Command-C it is not. This is not a problem because users can use the right click menu Copy function. This is just a minor problem. On MacOS we have other problems with key controls, for example, I discovered that  the Delete key is not working on Text Editor.

Reply:
Parser log is a copy of RIDE Log. The Select All keyboard shortcut is working on both windows, but the Command-C it is not. This is not a problem because users can use the right click menu Copy function. This is just a minor problem. On MacOS we have other problems with key controls, for example, I discovered that  the Delete key is not working on Text Editor.